packages/@aws-cdk/aws-applicationsignals-alpha/lib/slo/metric.ts

```diff
@@ -0,0 +1,357 @@
+import { ComparisonOperator, MetricStatistic, MetricType } from './constants';
+import { KeyAttributes } from './keyAttributes';
+
+/**
+ * Interface for metric dimension
+ */
+export interface MetricDimension {
+    /**
+     * Name of the dimension
+     *
+     * @required
+     */
+    readonly name: string;
+
+    /**
+     * Value of the dimension
+     *
+     * @required
+     */
+    readonly value: string;
+}
+
+/**
+ * Properties for Application Signals metrics
+ */
+export interface ApplicationSignalsMetricProps {
+    /**
+     * Type of the metric
+     *
+     * @required
+     */
+    readonly metricType: MetricType;
+
+    /**
+     * Key attributes for the service
+     *
+     * @required
+     */
+    readonly keyAttributes: KeyAttributes;
+
+    /**
+     * Operation name
+     *
+     * @default - undefined
+     */
+    readonly operationName?: string;
+
+    /**
+     * Period in seconds
+     * Required for period-based SLOs
+     *
+     * @required
+     */
+    readonly periodSeconds: number;
+
+    /**
+     * Statistic to use
+     * Required for period-based SLOs
+     *
+     * @required
+     */
+    readonly statistic: string;
+}
+
+/**
+ * Properties for CloudWatch metrics
+ */
+export interface CloudWatchMetricProps {
+    /**
+     * Metric data queries
+     *
+     * @required
+     */
+    readonly metricDataQueries: MetricDataQuery[];
+
+    /**
+     * Period in seconds
+     * Required for period-based SLOs
+     *
+     * @required
+     */
+    readonly periodSeconds: number;
+
+    /**
+     * Statistic to use
+     * Required for period-based SLOs
+     *
+     * @required
+     */
+    readonly statistic: string;
+}
+
+/**
+ * Properties for SLI metric configuration
+ */
+export type SliMetricBaseProps = {
+    /**
+     * The threshold value for the metric
+     *
+     * @required
+     */
+    readonly metricThreshold: number;
+
+    /**
+     * The comparison operator
+     *
+     * @default - Based on metric type
+     */
+    readonly comparisonOperator?: ComparisonOperator;
+} & (ApplicationSignalsMetricProps | CloudWatchMetricProps);
```

Comment:
Create actual interfaces, we do not support this while translating these types to other languages using JSII.